winml/adapter/winml_adapter_c_api.h

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
```

Comment:
Add comments to all of these methods to indicate why they exist and why we have not used the default c-abi implementation.